djangocms_picture/models.py

```diff
@@ -1,83 +1,236 @@
+        null=True,
+        on_delete=models.SET_NULL,
+        help_text=_('Wraps the image in a link to an internal (page) URL.'),
+    )
+    link_target = models.CharField(
+        verbose_name=_('Link target'),
+        choices=LINK_TARGET,
+        blank=True,
+        max_length=255,
+    )
+    link_attributes = AttributesField(
+        verbose_name=_('Link attributes'),
+        blank=True,
+        excluded_keys=['href', 'target'],
+    )
+    # cropping models
+    # active per default
+    use_automatic_scaling = models.BooleanField(
+        verbose_name=_('Automatic scaling'),
+        blank=True,
+        default=True,
+        help_text=_('Uses the placeholder dimenstions to automatically calculate the size.'),
+    )
+    # ignores all other cropping options
+    # throws validation error if other cropping options are selected
+    use_no_cropping = models.BooleanField(
+        verbose_name=_('Use original image.'),
+        blank=True,
+        default=False,
+        help_text=_('Outputs the raw image without cropping.'),
+    )
+    # upscale and crop work together
+    # throws validation error if other cropping options are selected
+    use_crop = models.BooleanField(
+        verbose_name=_('Crop image'),
+        blank=True,
+        default=False,
+        help_text=_('Crops the image according to the given thumbnail settings in the template.'),
+    )
+    use_upscale = models.BooleanField(
+        verbose_name=_('Upscale image'),
+        blank=True,
+        default=False,
+        help_text=_('Upscales the image to the size of the thumbnail settings in the template.')
+    )
+    # overrides all other options
+    # throws validation error if other cropping options are selected
+    use_thumbnail = models.ForeignKey(
+        ThumbnailOption,
+        verbose_name=_('Thumbnail options'),
+        blank=True,
+        null=True,
+        help_text=_('Overrides width, height, crop and upscale with the provided preset.'),
+    )
 
-    page_link = models.ForeignKey(
-        Page, verbose_name=_("page"), null=True,
-        limit_choices_to={'publisher_is_draft': True}, blank=True,
-        help_text=_("If present, clicking on image will take user to "
-                    "specified page."))
+    # Add an app namespace to related_name to avoid field name clashes
+    # with any other plugins that have a field with the same name as the
+    # lowercase of the class name of this model.
+    # https://github.com/divio/django-cms/issues/5030
+    cmsplugin_ptr = models.OneToOneField(
+        CMSPlugin,
+        related_name='%(app_label)s_%(class)s',
+        parent_link=True,
+    )
 
-    alt = models.CharField(
-        _("alternate text"), max_length=255, blank=True, null=True,
-        help_text=_("Specifies an alternate text for an image, if the image"
-                    "cannot be displayed.<br />Is also used by search engines"
-                    "to classify the image."))
+    def __str__(self):
```

Comment:
separate end user display from internals.
this is an internal, as such I'd let it just return the pk of the object which the cms already does.
instead define the `get_short_description()` method.